client/src/pages/queries/components/histories-tab.tsx

```diff
@@ -59,7 +59,7 @@ export default function HistoriesTab() {
       )
     }
 
-    if (data && data.length > 0) {
+    if (data && data[0].items.length) {
       // using reduce to flatten array from 2d to 1d
       const history = data
         .map((page) => page.items)
```

Comment:
We can use `flatMap` to convert a 2d array to 1d array or `data.flat()` would flatten the array